tools/uninstall.sh

```diff
@@ -0,0 +1,183 @@
+		$__result_var="no";
+		return 0;
+	elif [ "$answer_all_yes" == "true" ]; then
+		eval $__result_var="yes";
+		return 0;
+	fi
+	while true; do
+		read -p "$__prompt_text [Y/n] " _response
+		case $_response in
+			"" ) eval $__result_var="yes"; break;;
+			[Yy]* ) eval $__result_var="yes"; break;;
+			[Nn]* ) eval $__result_var="no";  break;;
+			* ) echo "Please answer yes or no.";;
+		esac
+	done
+	return 0;
+}
+
+function show_help {
+	echo "Usage: $(basename $0) [-h] [-n/-y]"
+	echo ""
+	echo "King Phisher Install Script"
+	echo ""
+	echo "optional arguments"
+	echo "  -h, --help			show this help message and exit"
+	echo "  -n, --no			 answer no to all questions"
+	echo "  -y, --yes			 answer yes to all questions"
+	echo "  --delete-database	 deletes king-phisher database and all campaigns"
+	echo "  --delete-directory	removes king-phisher directory and all files"
+	return 0;
+}
+
+while :; do
+	case $1 in
+		-h|-\?|--help)
+			show_help
+			exit
+			;;
+		-n|--no)
+			if [ "$answer_all_yes" == "true" ]; then
+				echo "Can not use -n and -y together"
+				exit $E_USAGE
+			fi
+			answer_all_no=true
+			;;
+		-y|--yes)
+			if [ "$answer_all_no" == "true" ]; then
+				echo "Can not use -n and -y together"
+				exit $E_USAGE
+			fi
+			answer_all_yes=true
+			;;
+		--delete-database)
+			KING_PHISHER_DELETE_DATABASE="x"
+			;;
+		--delete-dirctory)
+			KING_PHISHER_DELETE_DIRECTORY="x"
+			;;
+		--)
+			shift
+			break
+			;;
+		-?*)
+			printf "Unknown option: %s\n" "$1" >&2
+			exit $E_USAGE
+			;;
+		*)
+			break
+	esac
+	shift
+done
+
+if [ "$(id -u)" != "0" ]; then
+	echo "This must be run as root"
+	exit $E_NOTROOT
+fi
+
+
+if [ -f /lib/systemd/system/king-phisher.service ]; then
+	systemctl stop king-phisher
+	rm /lib/systemd/system/king-phisher.service
+	if [ -f /etc/systemd/system/multi-user.target.wants/king-phisher.service ]; then
+		rm /etc/systemd/system/multi-user.target.wants/king-phisher.service
+	fi
+	systemctl daemon-reload
+fi
+
+if [ -f /etc/init/king-phisher.conf ]; then
+	service king-phisher stop
+	rm /etc/init/king-phisher.conf
+fi
+
+if [ -f "/usr/local/share/applications/king-phisher.desktop" ]; then
+	rm /usr/local/share/applications/king-phisher.desktop
+	echo "removed king-phisher.desktop"
+elif [ -f "/usr/share/applications/king-phisher.desktop" ]; then
+	rm /usr/share/applications/king-phisher.desktop
+	echo "removed king-phisher.desktop"
+fi
+if [ -f /usr/share/icons/hicolor/scalable/apps/king-phisher-icon.svg ]; then
+	rm /usr/share/icons/hicolor/scalable/apps/king-phisher-icon.svg
+	echo "removed king-phisher icon"
+fi
+
+if [ -f "/usr/share/icons/hicolor/index.theme" -a "$(command -v gtk-update-icon-cache)" ]; then
+	echo "Updating the GTK icon cache"
+	gtk-update-icon-cache --force /usr/share/icons/hicolor
+fi
+
+if [ $answer_all_yes == "true" ]; then
+	KING_PHISHER_DELETE_DATABASE="x"
+	KING_PHISHER_DELETE_DIRECTORY="x"
+	answer_all_yes=false # reset to allow for last chance
```

Comment:
Also if the user specifies yes for all then there probably shouldn't be a confirmation. Otherwise the option is misleading.